mm2src/mm2_io/src/fs.rs

```diff
@@ -276,10 +271,56 @@ where
     read_files_with_extension(dir_path, "json").await
 }
 
+/// Creates all the directories along the path to a file if they do not exist.
+pub fn create_parents(path: &dyn AsRef<Path>) -> IoResult<()> {
+    let parent_dir = path.as_ref().parent();
+    let Some(parent_dir) = parent_dir else {
+        return MmError::err(
+            io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("{} has no parent directory", path.as_ref().display()),
+            ))
+    };
+    if parent_dir.exists() {
+        if !parent_dir.is_dir() {
+            return MmError::err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("{} is not a directory", parent_dir.display()),
+            ));
+        }
+    } else {
+        fs::create_dir_all(parent_dir)?;
+    }
+    Ok(())
+}
+
+/// Writes the `content` to the file at `path`.
+///
+/// This also creates any intermediary directories up to the file itself if they do not exist.
+/// If `use_tmp_file` is true, it writes to a temporary file first and then renames it to the final file name
+/// to ensure atomicity.
+pub fn write(path: &dyn AsRef<Path>, content: &[u8], use_tmp_file: bool) -> IoResult<()> {
+    // Create all the directories in the path.
+    create_parents(path)?;
+    let path_tmp = if use_tmp_file {
+        PathBuf::from(format!("{}.tmp", path.as_ref().display()))
+    } else {
+        path.as_ref().to_path_buf()
+    };
+    // Write the file content into the temp file and then rename the temp file into the desired name.
+    fs::write(&path_tmp, content)?;
+    if use_tmp_file {
+        fs::rename(&path_tmp, path.as_ref()).error_log_passthrough()?
+    }
+    Ok(())
+}
```

Comment:
I strongly believe that `create_parents` is redundant function and `write` should be renamed to avoid confusion with simple writing logic. But I suppose you disagree. Feel free to ignore.